File: front/shared/types/feedback.ts
export type MessageFeedbackRating = 'helpful' | 'not_helpful'
export type MessageFeedbackEvidenceQuality = 'sufficient' | 'insufficient' | 'missing'

export interface MessageFeedbackUpsertRequest {
  rating: MessageFeedbackRating
  evidence_quality?: MessageFeedbackEvidenceQuality | null
  hallucination_flag?: boolean
  comment?: string | null
}

export interface MessageFeedbackResponse {
  id: string
  user_id: string
  conversation_id: string
  message_id: string
  run_id?: string | null
  rating: MessageFeedbackRating
  evidence_quality?: MessageFeedbackEvidenceQuality | null
  hallucination_flag: boolean
  comment?: string | null
  created_at: string
  updated_at: string
}

export interface AdminMessageFeedbackItem {
  id: string
  user_id: string
  user_phone: string
  conversation_id: string
  message_id: string
  run_id?: string | null
  rating: MessageFeedbackRating
  evidence_quality?: MessageFeedbackEvidenceQuality | null
  hallucination_flag: boolean
  comment?: string | null
  message_preview: string
  created_at: string
  updated_at: string
}

export interface AdminMessageFeedbackListResponse {
  items: AdminMessageFeedbackItem[]
  total: number
  page: number
  page_size: number
}

export interface AdminMessageFeedbackSummaryResponse {
  total_feedback: number
  helpful_count: number
  not_helpful_count: number
  hallucination_count: number
  evidence_missing_count: number
  evidence_insufficient_count: number
  comment_count: number
}

File: app/schemas/feedback.py
"""Schemas for user answer feedback."""

from __future__ import annotations

from datetime import datetime
from typing import Literal
from uuid import UUID

from pydantic import BaseModel, Field

FeedbackRating = Literal["helpful", "not_helpful"]
FeedbackEvidenceQuality = Literal["sufficient", "insufficient", "missing"]


class MessageFeedbackUpsertRequest(BaseModel):
    rating: FeedbackRating
    evidence_quality: FeedbackEvidenceQuality | None = None
    hallucination_flag: bool = False
    comment: str | None = Field(default=None, max_length=2000)


class MessageFeedbackResponse(BaseModel):
    id: UUID
    user_id: UUID
    conversation_id: UUID
    message_id: UUID
    run_id: UUID | None = None
    rating: FeedbackRating
    evidence_quality: FeedbackEvidenceQuality | None = None
    hallucination_flag: bool
    comment: str | None = None
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}


class AdminMessageFeedbackItem(BaseModel):
    id: UUID
    user_id: UUID
    user_phone: str
    conversation_id: UUID
    message_id: UUID
    run_id: UUID | None = None
    rating: FeedbackRating
    evidence_quality: FeedbackEvidenceQuality | None = None
    hallucination_flag: bool
    comment: str | None = None
    message_preview: str
    created_at: datetime
    updated_at: datetime


class AdminMessageFeedbackListResponse(BaseModel):
    items: list[AdminMessageFeedbackItem]
    total: int
    page: int
    page_size: int


class AdminMessageFeedbackSummaryResponse(BaseModel):
    total_feedback: int = 0
    helpful_count: int = 0
    not_helpful_count: int = 0
    hallucination_count: int = 0
    evidence_missing_count: int = 0
    evidence_insufficient_count: int = 0
    comment_count: int = 0
